client/src/components/MapView.vue

```diff
@@ -208,6 +207,13 @@ export default defineComponent({
 
       this.setUpInteractionHandlers();
       this.setUpControls();
+
+      // Check whether there's a layer selected in the router.
+      if (router.currentRoute.value.query?.layer != null) {
+        this.state.currentDataLayer = this.state.dataLayers.find(l => l.styleLayer.id == router.currentRoute.value.query.layer) ?? leadServiceLinesByWaterSystemLayer;
```

Comment:
nit: setCurrentDataLayer()

Reply:
Thank you for this suggestion! Just saw it on your PR